lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -162,7 +172,7 @@ class GoogleCloudOutput < BufferedOutput
     attr_reader :running_on_managed_vm
     attr_reader :gae_backend_name
     attr_reader :gae_backend_version
-    attr_reader :service_name
+    attr_reader :resource_type
```

Comment:
The `merge_into_self` method is never registered in the C implementation:
https://github.com/google/protobuf/blob/master/ruby/ext/google/protobuf_c/map.c#L808
Thus we can't easily patch the method at Ruby level I'm afraid.

And if I use `merge` instead of `merge!`, it would hurt the performance, which is not what we want either, right?